test/hotspot/jtreg/compiler/gcbarriers/TestImplicitNullChecks.java

```diff
@@ -33,21 +33,11 @@
 import jdk.test.lib.Asserts;
 
 /**
- * @test id=G1
- * @summary Test that implicit null checks are generated as expected for G1
- *          memory accesses with barriers.
+ * @test
+ * @summary Test that implicit null checks are generated as expected for
+            different GC memory accesses.
  * @library /test/lib /
- * @requires vm.gc.G1
- * @run driver compiler.gcbarriers.TestImplicitNullChecks G1
- */
-
-/**
- * @test id=Z
- * @summary Test that implicit null checks are generated as expected for ZGC
-            memory accesses with barriers.
- * @library /test/lib /
- * @requires vm.gc.Z
- * @run driver compiler.gcbarriers.TestImplicitNullChecks Z
+ * @run driver compiler.gcbarriers.TestImplicitNullChecks
```

Comment:
I suppose you could still have a special run with `ZGC` and one with `G1GC`. But not sure if that is worth it, or if we do that in higher tiers anyway?

Reply:
> But not sure if that is worth it, or if we do that in higher tiers anyway?

We already run this test with G1 (default) on tier1 and with all non-default GCs (including ZGC) in Oracle's internal tier3.